packages/perspective-viewer-d3fc/src/js/charts/treemap.js

```diff
@@ -0,0 +1,64 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+
+import * as d3 from "d3";
+import {treeColor} from "../series/treemap/treemapColor";
+import {treemapSeries} from "../series/treemap/treemapSeries";
+import {tooltip} from "../tooltip/tooltip";
+import {gridLayoutMultiChart} from "../layout/gridLayoutMultiChart";
+
+function treemap(container, settings) {
+    const treemapGrid = gridLayoutMultiChart()
+        .elementsPrefix("treemap")
+        .treeColor(treeColor)
```

Comment:
This is feeling like it could be simplified. We're having to pass in `treeColor`, which is a function for creating the actual color, along with `treeColorDataMap` to help do that. Meanwhile, it's perhaps not obvious that the `gridLayoutMultiChart` component itself is processing the data and binding it with the DOM.

As a suggestion, which may help and could be worth trying: Perhaps pull the data function out of the `gridLayoutMultiChart` component (`const data = treeData(settings);`). Instead, do that in the calling class then attach it to the component using D3's `datum`:

```
container.datum(data).call(sunburstGrid);
```

You can then calculate the color in the calling class too, and pass in one thing instead of two functions to create it. You may be able to eliminate passing in `settings` as well.

Reply:
Okay, I can see what you mean. Will have another look.